### PR TITLE
Make KeyBindingInputManagers into Containers

### PR DESCRIPTION
**Pull request by peppy**
It turns out the overhead and complexity is a lot lower this way.

This resolves the inability to nest `KeyBinding` handlers inside each other and better handles cross-`Container` and cross-`InputManager` scenarios.

This removes the workaround in `TextBox` which involved every `TextBox` having their own `InputManager`.

### CODE REVIEW

**Comment by peppy** [comment]
Mouse wheel currently not working as intended. Will need a fix for that before merging.

**Comment by peppy** [comment]
Said mouse wheel issue has now been resolved 👍

File: osu.Framework/Platform/GameHost.cs [hunk 1]
```diff
@@ -406,7 +406,16 @@ private void resetInputHandlers()
...
         {
-            var root = new UserInputManager { Child = game };
+            var root = new UserInputManager
+            {
+                Child = new FrameworkActionContainer
+                {
+                    Child = new PlatformActionContainer
+                    {
+                        Child = game
+                    }
```
Comment (smoogipoo) [inline osu.Framework/Platform/GameHost.cs]: Isn't the FrameworkActionContainer supposed to be the child of PlatformActionContainer?

File: osu.Framework/Input/Bindings/KeyBindingContainer.cs [hunk 2]
```diff
@@ -188,12 +200,16 @@ protected virtual bool PropagateReleased(IEnumerable<Drawable> drawables, T rele
 
             return handled != null;
         }
+
+        public void TriggerReleased(T released) => PropagateReleased(KeyBindingInputQueue, released);
+
+        public void TriggerPressed(T pressed) => PropagateReleased(KeyBindingInputQueue, pressed);
```
Comment (smoogipoo) [inline osu.Framework/Input/Bindings/KeyBindingContainer.cs]: TriggerPressed propagates released? Re-test to make sure that gameplay works.

Comment (smoogipoo) [reply osu.Framework/Input/Bindings/KeyBindingContainer.cs]: Or I guess replays in this case >_>

**Comment by smoogipoo** [comment]
Looks good apart from mentioned issues. 👍

File: osu.Framework/Input/Bindings/KeyBindingContainer.cs (hunk 2 above)
Comment (peppy) [reply osu.Framework/Input/Bindings/KeyBindingContainer.cs]: looks like a typo... not sure how this even worked.

File: osu.Framework/Platform/GameHost.cs (hunk 1 above)
Comment (peppy) [reply osu.Framework/Platform/GameHost.cs]: They are independent, so order doesn't really matter (apart from which takes precedence). I think switching them around makes sense.

**Pull request merged**